pkg/api/machine.go

```diff
@@ -107,3 +121,8 @@ const (
 	NetworkInterfaceStatePending  NetworkInterfaceState = "Pending"
 	NetworkInterfaceStateAttached NetworkInterfaceState = "Attached"
 )
+
+type AgentStatus struct {
+	Type GuestAgent `json:"type"`
```

Comment:
Does this bring us any benefit having this here again? If no: I prefer to have the `AgentStatus` as a pointer in the `MachineStatus`

Reply:
No significant benefits, we can say you don't need handle nil for GuestAgent in status in future. I can revert it.

Reply:
fixed

Reply:
```suggestion
```